chisel/src/main.rs

```diff
@@ -0,0 +1,77 @@
+extern crate ansi_term;
+extern crate clap;
+extern crate libchisel;
+extern crate parity_wasm;
+extern crate toml;
+
+use std::process;
+use std::fs::File;
+use std::io;
+use std::io::prelude::*;
+
+use clap::{App, Arg};
+use ansi_term::Colour::Red;
+use libchisel::{ModuleCreator, ModuleTranslator, ModuleValidator};
+use parity_wasm::elements::*;
+use toml::Value;
+
+static ERR_PREFIX: &'static str = "error: ";
+static ERR_OPEN_FILE: &'static str = "Failed to open file";
+static ERR_PARSE_TOML: &'static str = "Failed to parse TOML configuration";
+static ERR_DESERIALIZE_WASM: &'static str = "Failed to deserialize wasm module";
+
+static PROGRAM_VERSION: &'static str = "0.1.0";
+
+//TODO: maybe use external log crate for this and maybe a better workaround
+/// Simply leaves a pretty message and exits.
+fn err_exit(msg: &str) -> ! {
+    println!("{}{}", Red.bold().paint(ERR_PREFIX), msg);
+    process::exit(-1);
+}
+
+fn main() {
+    let arg_matches = App::new("chisel") 
+        .version(PROGRAM_VERSION)
+        .author("Jake Lang <jak3lang@gmail.com>")
+        .about("Command line frontend for wasm-chisel")
+        .arg(Arg::with_name("config")
+            .short("c")
+            .long("config")
+            .value_name("FILE")
+            .help("Sets a configuration file for wasm-chisel")
+            .takes_value(true))
+        .arg(Arg::with_name("INPUT")
+            .help("Sets the wasm binary to be chiseled"))
+        .get_matches();
+
+    let mut bytecode = Vec::new();
+
+    // Load bytecode from file, if provided
+    if let Some(file) = arg_matches.value_of("INPUT") {
+        File::open(file)
+            .unwrap_or_else(|_x| err_exit(ERR_OPEN_FILE))
+            .read_to_end(&mut bytecode)
+            .expect("Could not read into buffer");
+    } else {
+        io::stdin().read_to_end(&mut bytecode) // If no file is specified, read from stdin
+        .expect("Could not read from stdin");
+    };
+
+    // Parse TOML configuration if provided
+    let config_toml = if let Some(file) = arg_matches.value_of("config") {
+        let mut toml_raw = String::new();
+        File::open(file)
+            .unwrap_or_else(|_x| err_exit(ERR_OPEN_FILE))
+            .read_to_string(&mut toml_raw)
+            .expect("Could not read into buffer"); 
+        Some(toml_raw.parse::<Value>().unwrap_or_else(|_x| err_exit(ERR_PARSE_TOML)))
```

Comment:
Decided to use unstructured TOML parsing for now instead of making a giant mess of structs with serde. This can be changed later but for brevity it works.